### ISSUE TITLE
CSRFCheck does not respect configured handler

### ISSUE THREAD

**Issue by denisname** [playframework/playframework#8971]
In Play 2.6 with Scala, the `play.filters.csrf.CSRFCheck.apply` method, when not given a handler, does not respect `play.filters.csrf.errorHandler` configured value and always default to `CSRF.DefaultErrorHandler`. See [CSRFActions.scala](https://github.com/playframework/playframework/blob/8793cfaa568f79e822cb11c1d6a156331ca9ba74/framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala#L566-L577).

**Reproducible Test Case**

In `conf/application.conf`:
> play.filters.csrf.errorHandler = controllers.CSRFErrorHandler

In `app/controllers/Application.scala`
```Scala
package controllers

import javax.inject.Inject
import scala.concurrent._
import play.api.mvc._
import play.api.mvc.Results._
import play.filters.csrf.CSRF
import play.filters.csrf.CSRFCheck

class AppController @Inject() (
  val controllerComponents: ControllerComponents,
  val checkToken: CSRFCheck
) extends BaseController {

  // Says "CSRF token check failed" expected "My CSFRErrorHandler".
  def test = checkToken(Action {
    Ok("Test")
  })

  // Says "My CSFRErrorHandler" as expected.
  def testWithHandler = checkToken(Action {
    Ok("Test")
  }, new CSRFErrorHandler())
}

class CSRFErrorHandler extends CSRF.ErrorHandler {
  def handle(request: RequestHeader, msg: String) = {
    Future.successful(Forbidden("My CSFRErrorHandler"))
  }
}
```

**Comment by matthewmascord** [comment]
Has this issue been fixed in Play 2.7 and above?

**Comment by mkurz** [comment]
Finally a fix is here (was very obvious actually, just no one came up with it until now...)
- #12481
